pkg/apis/sources/v1alpha2/ping_validation.go

```diff
@@ -40,3 +43,16 @@ func (cs *PingSourceSpec) Validate(ctx context.Context) *apis.FieldError {
 	}
 	return errs
 }
+
+func ValidateAnnotations(errs *apis.FieldError, annotations map[string]string) *apis.FieldError {
+	if annotations != nil {
+		if scope, ok := annotations[eventing.ScopeAnnotationKey]; ok {
+			if scope != "resource" && scope != "cluster" {
```

Comment:
Could we hoist these constants to somewhere like:
pkg/apis/sources/register.go?
and, maybe add the validationg logic there? At least I think I've seen these now in a few places.